### PR TITLE
Refactor Auth

### LINKED ISSUE #7: Refactor Authentication
- Use authentication library instead of custom implementation

### PR DESCRIPTION
**Pull request by HeathNaylor**
- dot env implemented
 - refactor old auth for sentinel
 - implemented Eloquent ORM

Closes #7

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,5 @@
 vendor
 .vagrant
 ubuntu-xenial-16.04-cloudimg-console.log
-.htaccess
\ No newline at end of file
+.htaccess
+.env
```
Comment (HeathNaylor) [inline .gitignore]: missing empty line at the end, configure your editor to do this automagically.

File: App/Config/sentinelConfig.php [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    /*
+    |--------------------------------------------------------------------------
+    | Session Key
+    |--------------------------------------------------------------------------
+    |
+    | Please provide your session key for Sentinel.
+    |
+    */
+
+    'session' => 'cartalyst_sentinel',
```
Comment (HeathNaylor) [inline App/Config/sentinelConfig.php]: lets give our session a name that reflects our application.

File: App/Config/sentinelConfig.php [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    /*
+    |--------------------------------------------------------------------------
+    | Cookie Key
+    |--------------------------------------------------------------------------
+    |
+    | Please provide your cookie key for Sentinel.
+    |
+    */
+
+    'cookie' => 'cartalyst_sentinel',
```
Comment (HeathNaylor) [inline App/Config/sentinelConfig.php]: same goes for here, keep the cookie and session name the same.

File: App/Config/sentinelConfig.php [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    | Roles
+    |--------------------------------------------------------------------------
+    |
+    | Please provide the role model used in Sentinel.
+    |
+    */
+
+    'roles' => [
+
+        'model' => 'Cartalyst\Sentinel\Roles\EloquentRole',
```
Comment (HeathNaylor) [inline App/Config/sentinelConfig.php]: Lets go ahead and be consistent, use the php static class method reference on `Cartalyst\Sentinel\Roles\EloquentRole`.

```php
'model' => \Cartalyst\Sentinel\Roles\EloquentRole::class,
```

File: App/Config/sentinelConfig.php [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    | permissions set on the role.
+    |
+    | "StrictPermissions" will deny any permission as soon as it finds it
+    | rejected on either the user or any of the assigned roles.
+    |
+    */
+
+    'permissions' => [
+
+        'class' => 'Cartalyst\Sentinel\Permissions\StandardPermissions',
```
Comment (HeathNaylor) [inline App/Config/sentinelConfig.php]: same goes for here, use static class method.

File: App/Config/sentinelConfig.php [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+    |--------------------------------------------------------------------------
+    |
+    | Here you may specify the persistences model used and weather to use the
+    | single persistence mode.
+    |
+    */
+
+    'persistences' => [
+
+        'model' => 'Cartalyst\Sentinel\Persistences\EloquentPersistence',
```
Comment (HeathNaylor) [inline App/Config/sentinelConfig.php]: static class method

File: App/Config/sentinelConfig.php [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+    | Here you may specify the activations model used and the time (in seconds)
+    | which activation codes expire. By default, activation codes expire after
+    | three days. The lottery is used for garbage collection, expired
+    | codes will be cleared automatically based on the provided odds.
+    |
+    */
+
+    'activations' => [
+
+        'model' => 'Cartalyst\Sentinel\Activations\EloquentActivation',
```
Comment (HeathNaylor) [inline App/Config/sentinelConfig.php]: static class method

File: App/Config/sentinelConfig.php [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+    | Here you may specify the reminders model used and the time (in seconds)
+    | which reminder codes expire. By default, reminder codes expire
+    | after four hours. The lottery is used for garbage collection, expired
+    | codes will be cleared automatically based on the provided odds.
+    |
+    */
+
+    'reminders' => [
+
+        'model' => 'Cartalyst\Sentinel\Reminders\EloquentReminder',
```
Comment (HeathNaylor) [inline App/Config/sentinelConfig.php]: static class method

File: App/Config/sentinelConfig.php [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+    | attempting to breach it could be an appropriate response.
+    |
+    | You may use any type of throttling for any scenario, and the specific
+    | configurations are designed to be customized as your site grows.
+    |
+    */
+
+    'throttling' => [
+
+        'model' => 'Cartalyst\Sentinel\Throttling\EloquentThrottle',
```
Comment (HeathNaylor) [inline App/Config/sentinelConfig.php]: static class method

File: App/Services/AuthenticateService.php [hunk 10]
```diff
@@ -2,12 +2,21 @@
...
 class AuthenticateService
 {
+    private $sentinel;
+
+    public function __construct()
+    {
+        $config = include BASEPATH . '/App/Config/sentinelConfig.php';
+        $sentinelBootstrapper = new SentinelBootstrapper($config);
+        $facade = new Sentinel($sentinelBootstrapper);
+        $this->sentinel = $facade->getSentinel();
```
Comment (HeathNaylor) [inline App/Services/AuthenticateService.php:20]: We are going to leverage dependency injection here. It is not the responsibility of the Auth service to bootstrap sentinel.

File: .gitignore (hunk 1 above)
Comment (Cisco89) [reply .gitignore]: fixed

File: App/Services/AuthenticateService.php (hunk 10 above)
Comment (Cisco89) [reply App/Services/AuthenticateService.php:20]: added a @todo to address this at adequate time

**Pull request merged**